app/controllers/concerns/api/v2/lookup_keys_common_controller.rb

```diff
@@ -125,6 +127,14 @@ def return_if_smart_mismatch
     end
   end
 
+  def cast_default_value
+    obj = smart_variable_id? ? "smart_variable" : "smart_class_parameter"
+
+    return unless params[obj] && params[obj].key?(:default_value) && !params[obj][:default_value].is_a?(String)
```

Comment:
This code is basically a duplicate of `cast_value` in override_values_controller.rb, please refactor to remove the duplication